os/arch/arm/src/imxrt/imxrt_edma.c

```diff
@@ -84,1242 +84,2299 @@
  *
- * Assumptions:
- *   Called early in the initialization sequence so no special protection is
- *   necessary.
+ * A minor offset is a signed-extended value added to the source address or a destination
+ * address after each minor loop.
  *
- ****************************************************************************/
-
-#if CONFIG_IMXRT_EDMA_NTCD > 0
-static inline void imxrt_tcd_initialize(void)
+ * param tcd A point to the TCD structure.
+ * param config A pointer to the minor offset configuration structure.
+ */
+void imxrt_edma_tcdsetminoroffsetconfig(edma_tcd_t *tcd, const edma_minor_offset_config_t *config)
 {
-	sq_entry_t *tcd;
-	int i;
+    assert(tcd != NULL);
+    assert(((uint32_t)tcd & 0x1FU) == 0);
 
-	/* Add each pre-allocated TCD to the tail of the TCD free list */
+    uint32_t tmpreg;
 
-	sq_init(&g_tcd_free);
-	for (i = 0; i < CONFIG_IMXRT_EDMA_NTCD; i++) {
-		tcd = (sq_entry_t *)&g_tcd_pool[i];
-		sq_addlast(tcd, &g_tcd_free);
-	}
+    tmpreg = tcd->NBYTES &
+             ~(DMA_NBYTES_MLOFFYES_SMLOE_MASK | DMA_NBYTES_MLOFFYES_DMLOE_MASK | DMA_NBYTES_MLOFFYES_MLOFF_MASK);
+    tmpreg |=
+        (DMA_NBYTES_MLOFFYES_SMLOE(config->enableSrcMinorOffset) |
+         DMA_NBYTES_MLOFFYES_DMLOE(config->enableDestMinorOffset) | DMA_NBYTES_MLOFFYES_MLOFF(config->minorOffset));
+    tcd->NBYTES = tmpreg;
 }
-#endif
 
-/************************************************************************************
- * Name: imxrt_tcd_chanlink
- *
- * Description:
- *   This function configures either a minor link or a major link. The minor link
- *   means the channel link is triggered every time CITER decreases by 1. The major
- *   link means that the channel link  is triggered when the CITER is exhausted.
+/*!
+ * brief Sets the channel link for the eDMA TCD.
  *
- *   NOTE: Users should ensure that DONE flag is cleared before calling this
- *   interface, or the configuration is invalid.
+ * This function configures either a minor link or a major link. The minor link means the channel link is
+ * triggered every time CITER decreases by 1. The major link means that the channel link  is triggered when the CITER is
+ * exhausted.
  *
- * Input Parameters:
- *   tcd  - Point to the TCD structure.
- *   type - Channel link type.
- *   chan - The linked channel number.
+ * note Users should ensure that DONE flag is cleared before calling this interface, or the configuration is invalid.
+ * param tcd Point to the TCD structure.
+ * param type Channel link type, it can be one of:
+ *   arg kEDMA_LinkNone
+ *   arg kEDMA_MinorLink
+ *   arg kEDMA_MajorLink
+ * param linkedChannel The linked channel number.
+ */
+void imxrt_edma_tcdsetchannellink(edma_tcd_t *tcd, edma_channel_link_type_t type, uint32_t linkedChannel)
+{
+    assert(tcd != NULL);
+    assert(((uint32_t)tcd & 0x1FU) == 0);
+    assert(linkedChannel < FSL_FEATURE_EDMA_MODULE_CHANNEL);
+
+    if (type == kEDMA_MinorLink) /* Minor link config */
+    {
```

Comment:
same as above, please do it in entire file

Reply:
I fixed it
Please check it